src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1624,34 +1624,52 @@ MavlinkReceiver::handle_message_hil_state_quaternion(mavlink_message_t *msg)
 void *
 MavlinkReceiver::receive_thread(void *arg)
 {
-#ifndef __PX4_POSIX
-	int uart_fd = _mavlink->get_uart_fd();
 
 	const int timeout = 500;
 	uint8_t buf[32];
-
 	mavlink_message_t msg;
 
-	/* set thread name */
-	char thread_name[24];
-	sprintf(thread_name, "mavlink_rcv_if%d", _mavlink->get_instance_id());
-	prctl(PR_SET_NAME, thread_name, getpid());
-
 	struct pollfd fds[1];
-	fds[0].fd = uart_fd;
-	fds[0].events = POLLIN;
+
+	struct sockaddr_in srcaddr;
+	socklen_t addrlen = sizeof(srcaddr);
+
+	int uart_fd;
+
+	if (_mavlink->get_protocol() == SERIAL) {
+		uart_fd = _mavlink->get_uart_fd();
+#ifndef __PX4_POSIX
```

Comment:
This needs to be abstracted into a px4_set_thread_name() function, which then has different files handling the different OS implementations. But that's for another time.